### PR TITLE
[IMP] web: grey out avatar of archived users/employees

### PR DESCRIPTION
**Pull request by hap-odoo**
The purpose of this commit is to let users know when
another user or employee has been archived.
This is done by differentiates the archived user or employee
by greying out the avatar of it.

TaskID-2514400

### CODE REVIEW

File: odoo/tools/image.py [hunk 1]
```diff
@@ -262,16 +263,16 @@ def colorize(self):
...
     return the result as a base64 encoded image.
     """
-    if not base64_source or ((not size or (not size[0] and not size[1])) and not verify_resolution and not quality and not crop and not colorize and not output_format):
+    if not base64_source or ((not size or (not size[0] and not size[1])) and not verify_resolution and not quality and not crop and not colorize and not output_format and user_active):
         # for performance: don't do anything if the image is falsy or if
         # no operations have been requested
         return base64_source
 
-    image = ImageProcess(base64_source, verify_resolution)
+    image = ImageProcess(base64_source, user_active, verify_resolution)
```
Comment (JKE-be) [inline odoo/tools/image.py:275]: `ImageOps.grayscale(image)`  should probably be done here, it is not linked to the `base64_to_image` function.
And param name be something related to grayScale and not user... Here your are in tools...

`image_base64 = image_process(image_base64, size=(int(width), int(height)), crop=crop, quality=int(quality), grayscale=not user_active)` for example

**Comment by mart-e** [comment]
Hi,
Can't we make this purely in CSS on the image widget? How about if I archive and unarchive a user?

**Comment by msh-odoo** [comment]
> Hi,
> Can't we make this purely in CSS on the image widget? https://developer.mozilla.org/en-US/docs/Web/CSS/filter-function/grayscale()
> Isn't that redundant with the ribbon widget?

The reason for doing code in controller instead of css is to save rpc calls, if we do it in css we need to add class to make image grey based on Archived/Unarchived value, we need to check first whether model has active field if yes then whether it's value is true/false, also if we do this code in avatar widget then we need to think about listview as avatar widget can be in listview where we may have hundreds of records, so we needed to handle reading of active field value wihtout doing extra rpc so we thought why not make imag grey when image controller is called.

**Comment by mart-e** [comment]
You don't necessarily need another rpc, it's just reading the field `active` if present, the same way we are already doing for web_ribbon widget. This should be generic, don't hardcode the models in `web` module, one use could want to display archived products in grey for instance.

**Comment by seb-odoo** [comment]
Hello, I think this task should be delayed and done after https://www.odoo.com/web#active_id=2404630&cids=1&id=2404630&model=project.task
This other task will introduce a new field specific for user avatar, where it will be easier to return a greyed version of it without having to further hack the current image route.

File: odoo/tools/image.py [hunk 2]
```diff
@@ -49,7 +49,7 @@
 
 class ImageProcess():
 
-    def __init__(self, base64_source, verify_resolution=True):
+    def __init__(self, base64_source, grayscale=False, verify_resolution=True):
```
Comment (seb-odoo) [inline odoo/tools/image.py:52]: grayscale should be an operation that you call in the chain, not a param of the constructor

File: odoo/tools/image.py [hunk 3]
```diff
@@ -389,7 +389,7 @@ def image_fix_orientation(image):
     return image
 
 
-def base64_to_image(base64_source):
+def base64_to_image(base64_source, grayscale=False):
```
Comment (seb-odoo) [inline odoo/tools/image.py:392]: This should be in the processing, not in the converter.

**Review by seb-odoo — CHANGES_REQUESTED** [review]

**Comment by msh-odoo** [comment]
> You don't necessarily need another rpc, it's just reading the field `active` if present, the same way we are already doing for web_ribbon widget. This should be generic, don't hardcode the models in `web` module, one use could want to display archived products in grey for instance.

@mart-e The proble here is we do not want to read active field of current model, we want to check comodel record's active field, we want to check if comodel's record is archived then we want to make image field gray, the task is to make Many2oneAvatar widget with gray image if many2one record is archived.

**Comment by C3POdoo** [comment]
Dear @hap-odoo,

Thank you for your contribution but we are closing it due to inactivity.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**